README.md

```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
 These are then exposed via a [gitcredential helper](https://git-scm.com/docs/gitcredentials) which will download the
 credentials as needed.
 
-### Environment variables
+## Options
+***
+The Vault Secrets plugin supports a number of different configuration options.
 
-Key values pairs can also be uploaded.
+### `server`
+The address of the target Vault server. Example: `https://my-vault-server:8200`
 
-```bash
-vault write data/buildkite/my_pipeline/environment value=- <<< $(echo "MY_SECRET=blah" | base64)
-```
+### `path`
+Alternative Base Path to use for Vault secrets. This is expected to be a [KV Store](https://developer.hashicorp.com/vault/docs/secrets/kv#kv-version-2)  
 
-```bash
-vault write data/buildkite/my_pipeline/env_key value=- <<< $(echo "my secret"| base64)
-```
+Defaults to: `data/buildkite`
 
-Can be loaded using:
+
+### `auth`
+Configure the authentication parameters the plugin should use to authenticate with Vault.
+
+`auth` accepts the following options:
+
+`method` (required)
+The auth method to use when authenticating with Vault. Currently only `approle` is supported
+
+`approle`
+Configures the plugin to use AppRole authentication to the Vault server. Requires a `role-id` be set.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`role-id` (required)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The role-id the plugin should use to authenticate to Vault
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`secret-env`
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
+
+
+
+Example:
 
 ```yaml
 steps:
   - command: ./run_build.sh
     plugins:
       - vault-secrets#v0.2.2:
-          server: my-vault-server
+          server: https://my-vault-server
           auth:
             method: 'approle'
             role-id: 'my-role-id'
-            secret-env: 'VAULT_SECRET_ID'
-          secrets:
-          - key
+            secret-env: 'MY_SECRET_ENV'
 ```
 
-## Options
-
-### `path`
-
-defaults to: `data/buildkite`
-This is expected to be a kv store
-
-Alternative Base Path to use for Vault secrets
-
 ## Testing
-
-To run locally:
-
+---
+### Unit tests
+The unit tests are written using BATS, you can test locally with:
 ```bash
-BUILDKITE_PLUGIN_VAULT_SECRETS_DUMP_ENV=true BUILDKITE_PLUGIN_VAULT_SECRETS_ADDR=http://0.0.0.0:8200 BUILDKITE_PIPELINE_SLUG=my_pipeline hooks/environment
+make test
 ```
-
-To test with BATS:
-
+or using docker-compose:
 ```bash
 docker-compose -f docker-compose.yml run --rm tests
 ```
 
-Integration test:
+### Integration test
+
+The integration tests are run by spinning up a local vault container in dev mode, and configuring them with some data.
 
 ```bash
-.buildkite/steps/test_envvar.sh
+make integration-test
 ```
 
 When writing test plans, note that secrets are processed in the order they appear in the list returned from the Vault.
 
-## TODO
+### Testing the pipeline
+You can test the pipeline locally using the `bk cli`
```

Comment:
I would add a clarification here about why the additional environment variable is necessary

Reply:
Good call - done!